rust/operator-binary/src/controller.rs

```diff
@@ -467,6 +468,47 @@ fn build_node_rolegroup_service(
     })
 }
 
+fn resolve_resource_config_for_rolegroup(
+    nifi: &NifiCluster,
+    rolegroup_ref: &RoleGroupRef<NifiCluster>,
+) -> Result<Resources<NifiStorageConfig, NoRuntimeLimits>> {
+    // Initialize the result with all default values as baseline
+    let conf_defaults = NifiConfig::default_resources();
+
+    // Retrieve global role resource config
+    let mut conf_role: Resources<NifiStorageConfig, NoRuntimeLimits> = nifi
+        .spec
+        .nodes
+        .as_ref()
+        .with_context(|| NoNodesDefinedSnafu {})?
+        .config
+        .config
+        .resources
+        .clone()
+        .unwrap_or_default();
+
+    // Retrieve rolegroup specific resource config
+    let mut conf_rolegroup: Resources<NifiStorageConfig, NoRuntimeLimits> = nifi
+        .spec
+        .nodes
+        .as_ref()
+        .context(NoNodeRoleSnafu)?
```

Comment:
Is this `NoNodeRole` or `NoNodesDefined`? Would it make sense to pull out the node role definition separately first before digging out the guts?

Reply:
Good catch with the error. I will check if an extracted role can be reused in other places too.

Reply:
The resources (and the role) are now computed once (per rolegroup).